docs/contributing.md

```diff
@@ -219,46 +219,67 @@ multi-threaded. It should use all the CPU cores without any configuration.
+The documentation for `jj` is automatically published online at
 <https://martinvonz.github.io/jj/>.
 
-When editing documentation, we'd appreciate it if you checked that the
-result will look as expected when published to the website.
+When editing documentation, you should check your changes locally — especially
+if you are adding a new page, or doing a major rewrite.
 
-### Setting up the prerequisites
+The only thing you need is [`uv`][uv], a Python project manager written in Rust.
+It will fetch the right Python version and the dependencies. Install it like so:
 
-To build the website, you must have `uv` installed. Please see: [Installing uv].
+[uv]: https://docs.astral.sh/uv/
 
-[Installing uv]: https://docs.astral.sh/uv/getting-started/installation/
+=== "macOS/Linux"
 
-Once you have `uv` installed, you should ask it to install the rest
-of the required tools into a virtual environment as follows:
+    ``` { .shell .copy }
+    curl -LsSf https://astral.sh/uv/install.sh | sh
+    ```
 
-```shell
-uv sync
-```
+=== "Windows"
 
-### Building the HTML docs locally (with live reload)
+    ``` { .shell .copy }
+    powershell -ExecutionPolicy ByPass -c "irm https://astral.sh/uv/install.ps1 | iex"
+    ```
 
-The HTML docs are built with [MkDocs](https://github.com/mkdocs/mkdocs). After
-following the above steps, you should be able to view the docs by running
+=== "Homebrew"
 
-```shell
-# Note: this and all the commands below should be run from the root of
-# the `jj` source tree.
+    ``` { .shell .copy }
+    brew install uv
+    ```
+
+=== "Cargo"
+
+    ``` { .shell .copy }
+    # This might take a while
+    cargo install --git https://github.com/astral-sh/uv uv
+    ```
+    
+=== "Other options"
+
+    See [Installing uv](https://docs.astral.sh/uv/getting-started/installation/).
+
+To build the docs, run from the root of the `jj` repository:
+
+``` { .shell .copy }
 uv run mkdocs serve
 ```
 
-and opening <http://127.0.0.1:8000> in your browser.
+Open <http://127.0.0.1:8000> in your browser to see the docs.
+
+As you edit the `.md` files in `docs/`, the website should be rebuilt and
+reloaded in your browser automatically.
 
-As you edit the `md` files, the website should be rebuilt and reloaded in your
-browser automatically, unless build errors occur.
+!!! note "If the docs are not updating"
+    Check the terminal from which you ran `uv run mkdocs serve` for any build
+    errors or warnings. Warnings about `"GET /versions.json HTTP/1.1" code 404`
+    are expected and harmless.
 
-You should occasionally check the terminal from which you ran `uv run mkdocs serve` for
-any build errors or warnings. Warnings about `"GET /versions.json HTTP/1.1" code
-404` are expected and harmless.
+## Building the entire website
 
-### How to build the entire website (not usually necessary)
+!!! tip
+    Building the entire website is not usually necessary. If you are editing
+    documentation, the previous section is enough.
```

Comment:
You could add: "These instructions are relevant if you are working on the versioning of the documentation that we currently do with `mike`."